src/tags/master_playlist/media.rs

```diff
@@ -126,9 +126,10 @@ impl ExtXMediaBuilder {
             return Err(Error::unexpected_attribute("INSTREAM-ID").to_string());
         }
 
-        if self.is_default.unwrap_or(false) && !self.is_autoselect.unwrap_or(false) {
```

Comment:
I think this line is correct:

> AUTOSELECT
> The value is an enumerated-string; valid strings are YES and NO.
> This attribute is OPTIONAL.  Its absence indicates an implicit
> value of NO.

The `Option` is to indicate, whether the value is specified and if in this case there is `None` it will be set to a `NO` (`false`)

Reply:
Do you have any real world examples, where this causes problems?

Reply:
I don't know the actual case related to that.
I just found the problem while testing your examples.
```
[hls_m3u8]$ cat examples/data/rfc8216_8-7.m3u8 | cargo run --example parse -- --m3u8-type master
   Compiling hls_m3u8 v0.2.1 (/home/ec2-user/workspace/sori/hls_m3u8)
    Finished dev [unoptimized + debuginfo] target(s) in 1.22s
     Running `target/debug/examples/parse --m3u8-type master`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner:

CustomError: BuilderError: CustomError: If `DEFAULT` is true, `AUTOSELECT` has to be true too! }', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

I considered many conditions based on a table
```
| DEFAULT  | DEFAULT  | AUTOSELECT | AUTOSELECT | rfc8216 |                            |          |
| PRESENT? | YES/NO   | PRESENT?   |  YES/NO    | VALID?  | RUST Expression            | INVALID? |
+----------+----------+------------+------------+---------+----------------------------+----------+
|   NO     |   .      |   NO       |   .        |   YES   | DEFAULT.wrap_or(false)     | false    |
|   YES    |   NO     |   YES      |   YES/NO   |   YES   | DEFAULT.wrap_or(false)     | false    |
|   YES    |   YES    |   NO       |   .        |   YES   | ! AUTOSELECT.wrap_or(true) | false    |
|   YES    |   YES    |   YES      |   YES      |   YES   | ! AUTOSELECT.wrap_or(true) | false    |
|   YES    |   YES    |   YES      |   NO       |   NO    | ! AUTOSELECT.wrap_or(true) | true     |

<=> VALID? == not(INVALID?)
```

The result is as follows:
```
[hls_m3u8]$ cat examples/data/rfc8216_8-7.m3u8 | cargo run --example parse -- --m3u8-type master
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/parse --m3u8-type master`
#EXTM3U
#EXT-X-MEDIA:TYPE=VIDEO,URI="low/main/audio-video.m3u8",GROUP-ID="low",NAME="Main",DEFAULT=YES
#EXT-X-MEDIA:TYPE=VIDEO,URI="low/centerfield/audio-video.m3u8",GROUP-ID="low",NAME="Centerfield"
#EXT-X-MEDIA:TYPE=VIDEO,URI="low/dugout/audio-video.m3u8",GROUP-ID="low",NAME="Dugout"
#EXT-X-MEDIA:TYPE=VIDEO,URI="mid/main/audio-video.m3u8",GROUP-ID="mid",NAME="Main",DEFAULT=YES
#EXT-X-MEDIA:TYPE=VIDEO,URI="mid/centerfield/audio-video.m3u8",GROUP-ID="mid",NAME="Centerfield"
#EXT-X-MEDIA:TYPE=VIDEO,URI="mid/dugout/audio-video.m3u8",GROUP-ID="mid",NAME="Dugout"
#EXT-X-MEDIA:TYPE=VIDEO,URI="hi/main/audio-video.m3u8",GROUP-ID="hi",NAME="Main",DEFAULT=YES
#EXT-X-MEDIA:TYPE=VIDEO,URI="hi/centerfield/audio-video.m3u8",GROUP-ID="hi",NAME="Centerfield"
#EXT-X-MEDIA:TYPE=VIDEO,URI="hi/dugout/audio-video.m3u8",GROUP-ID="hi",NAME="Dugout"
#EXT-X-STREAM-INF:BANDWIDTH=1280000,CODECS="...",VIDEO="low"
low/main/audio-video.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=2560000,CODECS="...",VIDEO="mid"
mid/main/audio-video.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=7680000,CODECS="...",VIDEO="hi"
hi/main/audio-video.m3u8
```